Deprecate `susemanager-frontend`

Etheryte opened the pull request:
## What does this PR change?

The main goal of this PR is to:  
 - Remove the `susemanager-frontend` separation and deprecate `susemanager-nodejs-sdk-devel`
 - To do that, migrate the packaging logic from [`susemanager-nodejs-sdk-devel.spec`](https://github.com/uyuni-project/uyuni/blob/549de8dd5847891a7f0f075ea1d630b849d7cdb9/susemanager-frontend/susemanager-nodejs-sdk-devel/susemanager-nodejs-sdk-devel.spec) into [`spacewalk-web.spec`](https://github.com/uyuni-project/uyuni/blob/51154d5f44df73cf59b40f1edd481bee75a4d842/web/spacewalk-web.spec)
 - Setup logic from [`susemanager-frontend/susemanager-nodejs-sdk-devel/setup.sh`](https://github.com/uyuni-project/uyuni/blob/549de8dd5847891a7f0f075ea1d630b849d7cdb9/susemanager-frontend/susemanager-nodejs-sdk-devel/setup.sh) into [`web/setup.sh`](https://github.com/uyuni-project/uyuni/blob/51154d5f44df73cf59b40f1edd481bee75a4d842/web/setup.sh)
 - And migrate frontend dependencies from [`susemanager-frontend/susemanager-nodejs-sdk-devel/package.json`](https://github.com/uyuni-project/uyuni/blob/549de8dd5847891a7f0f075ea1d630b849d7cdb9/susemanager-frontend/susemanager-nodejs-sdk-devel/package.json) into [`web/html/src/package.json`](https://github.com/uyuni-project/uyuni/blob/51154d5f44df73cf59b40f1edd481bee75a4d842/web/html/src/package.json)
 - (And update the Webpack build script to accommodate for these changes.)

The above changes will allow us to fix incorrect security alerts for Dependabot. (And also help prepare for https://github.com/SUSE/spacewalk/issues/13145, but that's not the focus here.)  

On the packaging side, the important change is that since `susemanager-nodejs-sdk-devel` no longer exists, `spacewalk-web` needs to provide the source `node-modules.tar.gz` (previously `susemanager-nodejs-modules.tar.gz`).  

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: build tooling.

- [x] **DONE**

## Test coverage
- No tests: build tooling.

- [x] **DONE**

## Links

Part of https://github.com/SUSE/spacewalk/issues/16424

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)


## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

Etheryte commented:
@agraul @juliogonzalez Sorry for the bother, if you have the time, could you please take a look and help me figure out why I can't get this to build correctly? The core of the related changes are to [`web/spacewalk-web.spec`](https://github.com/uyuni-project/uyuni/pull/4950/files#diff-325898a27431b9480c5b9155e098708b5e02388919ca187e0f9578183353bead) and to [`web/setup.sh`](https://github.com/uyuni-project/uyuni/pull/4950/files#diff-546e6f71180f889655e24bda285cd0ef1f861b1c113ca3a2ae5a9c72a9e87334).

juliogonzalez commented:
> could you please take a look and help me figure out why I can't get this to build correctly?

What's the error?

Etheryte commented:
The error is the one outlined in the PR description, when I try to use `rel-eng/build-packages-local-obs.sh` I get

```
RPM build errors:
Bad file: /tmp/push-packages-to-obs/SRPMBUILD/rpmbuild-spacewalk-webrw879qlk/SOURCES/node-modules.tar.gz: No such file or directory
```

I've outlined more details and what I've done exactly as well in the PR description, I just didn't want to ping you with a wall of text.

juliogonzalez commented:
> The error is the one outlined in the PR description, when I try to use `rel-eng/build-packages-local-obs.sh` I get
> 
> ```
> RPM build errors:
> Bad file: /tmp/push-packages-to-obs/SRPMBUILD/rpmbuild-spacewalk-webrw879qlk/SOURCES/node-modules.tar.gz: No such file or directory
> ```
> 
> I've outlined more details and what I've done exactly as well in the PR description, I just didn't want to ping you with a wall of text.

How was `susemanager-nodejs-modules.tar.gz`created in the past? `node-modules.tar.gz` should be created in the same way for `spacewalk-web`, so it's added to the sources.

This was some magic outside of my area of knowledge, maybe you need to check with Dario?

Etheryte commented:
As far as I know (but I could be wrong), there was no logic beyond what was in [`susemanager-frontend/susemanager-nodejs-sdk-devel/setup.sh`](https://github.com/uyuni-project/uyuni/pull/4950/files#diff-b23a8ca69910d18aaf19f8c7a770cbaf23461193dbe8d62e0cae0a3c9f2de360). However after I moved the logic over to [`web/setup.sh`](https://github.com/uyuni-project/uyuni/pull/4950/files#diff-546e6f71180f889655e24bda285cd0ef1f861b1c113ca3a2ae5a9c72a9e87334), I get no dice and the error shown above, even after I manually created the file and added it to my fork on OBS. @ncounter do you know what I could be doing wrong?

juliogonzalez commented:
Discussed at Slack. 

The tarball was created not at the `web/` but at `/web/html/src`. Adding a `mv` to `setup.sh` fixes the issue.

Etheryte commented:
This is ready for review. In short, the changes can be summarized as follows:  

 - `susemanager-frontend` along with `susemanager-nodejs-sdk-devel` no longer exists.  
 - All the dependencies and build logic was moved over to `spacewalk-web`.
 - `web/html/src/build.js` was updated to no longer require Webpack as a standalone executable.
 - Github actions etc were updated.

I have a build success on [my fork of spacewalk-web on obs](https://build.opensuse.org/package/show/home:Etheryte:branches:systemsmanagement:Uyuni:Master/spacewalk-web).  

Please let me know if there's anything I missed, or you feel is left over.

juliogonzalez commented on web/spacewalk-web.spec:
```diff
@@ -32,6 +32,7 @@
 %endif
 %endif
 %{!?rhel: %global sbinpath /sbin}%{?rhel: %global sbinpath %{_sbindir}}
+%{!?nodejs_sitelib:%define nodejs_sitelib %{_prefix}/lib/node_modules}
```
IIRC this PR correctly, you need to add the license from the old nodejs-sdk package to this package, right?

Also remember to remove the old nodejs-sdk packages from `rel-eng/packages`

juliogonzalez replied:
As similar to @agraul I just reviewed specs and changelogs, but not the rest of the changes, as I don't have enough knowledge to judge :-)

juliogonzalez reviewed (approved):
Should be OK now. Approved.

Etheryte commented:
Merging this now so I have time to address if any build issues come up before the end of the day.
